CMakeLists.txt

```diff
@@ -12,6 +12,7 @@ include(GNUInstallDirs)
 option(BUILD_GSTREAMER_PLUGIN "Build kvssink GStreamer plugin" OFF)
 option(BUILD_JNI "Build C++ wrapper for JNI to expose the functionality to Java/Android" OFF)
 option(BUILD_STATIC "Build with static linkage" OFF)
+option(BUILD_GSTREAMER_PLUGIN_STATIC "If building GStreamer plugin, build it as a static library" ${BUILD_STATIC})
```

Comment:
With existing implementation, if BUILD_STATIC is ON, then libgstkvssink is always built as a static library (which makes little sense for gstreamer usage). This new option allows to set BUILD_STATIC to ON, while building libgstkvssink as a shared library.
It's not perfect, but at least allows building libgstkvssink.so that has libcproducer built-in.

Reply:
The naming here is confusing and in the comments it says "If building GStreamer plugin, build it as a static library"

But what you're saying is you don't want to build it as a static library, rather you want to build as shared object and the shared object should in turn have it's producer c dependency statically linked in.  We should be explicit about what it does.  I am on board to have a build option that statically links in the producer library, but the plugin itself is still a shared object so we need the naming of such an option should be clear.